lib/rgeo/active_record/arel_spatial_queries.rb

```diff
@@ -41,13 +40,13 @@ def visit_RGeo_ActiveRecord_SpatialNamedFunction(node, collector)
       # The node must be a string (in which case it is treated as WKT),
       # an RGeo feature, or a spatial attribute.
       def visit_in_spatial_context(node, collector)
-        case node
-        when String
+        if node.is_a?(String)
```

Comment:
Yes, `case` is definitely more appropriate, but I had to use `is_a?` because of a bug in the RGeo core library.

Under the hood, `case` uses `#===` and this is evidently not implemented for `RGeo::Feature` or it is not working correctly. This example shows the bug:

```rb
pt = RGeo::Geos.factory.point(1,2)
#=> #<RGeo::Geos::CAPIPointImpl "POINT (1.0 2.0)"> 
pt === RGeo::Feature
#=> false
```

So I decided to use `is_a?` until we fix that bug in the core library.